obsidian.el

```diff
@@ -69,110 +68,292 @@
+  "Position of backlinks buffer in frame.
+Valid values are:
+ * `right',
+ * `left'."
+  :type '(choice (const right)
+                 (const left))
+  :group 'backlinks-window)
+
+(defcustom obsidian-backlinks-panel-width 75
+  "Width of the backlinks window."
+  :type 'integer
+  :group 'backlinks-window)
+
+(defcustom obsidian-backlinks-filename-proportion 1.0
+  "Proportion of space to be used to display the file.
+The remainder will be used to display the link text.
+Setting a value of 1.0+ will cause 2 lines to be used per backlink with
+the filename on the first line and the link text on the line below."
+  :type 'float
+  :group 'backlinks-window)
+
+(defcustom obsidian-backlinks-show-vault-path t
+  "If t, show path relative to Obsidian vault, otherwise only show file name."
+  :type 'boolean
+  :group 'backlinks-window)
+
+(defcustom obsidian-backlinks-buffer-name "*backlinks*"
+  "Name to use for the obsidian backlinks buffer."
+  :type 'string
+  :group 'backlinks-window)
+
+(defcustom obsidian-daily-note-template nil
+  "Daily notes' template filename in templates directory."
   :type 'file)
 
+(defcustom obsidian-use-update-timer t
+  "Determines whether a polling cache update will be used.
+If it is true, a timer will be created using the values of
+`obsidian-cache-expiry' and `obsidian-update-idle-wait'."
+  :type 'boolean)
+
+(defcustom obsidian-cache-expiry (* 60 5)
+  "The number of seconds before the Obsidian cache will update."
+  :type 'integer
+  :group 'obsidian)
+
+(defcustom obsidian-update-idle-wait 5
+  "Seconds to wait after cache expiry for Emacs to be idle before running update."
+  :type 'integer
+  :group 'obsidian)
+
 (eval-when-compile (defvar local-minor-modes))
 
-(defun obsidian--directory-files-pre28 (orig-func dir &optional full match nosort ignored)
+(defun obsidian--directory-files-pre28
+    (orig-func dir &optional full match nosort _)
   "Version of `directory-files' compatible with Emacs versions < 28.
 
 ORIG-FUNC is the original `directory-files' function that is going to be
 advised,and DIR and the directory of files on which `directory-files' will
 be called.
 FULL, MATCH, and NOSORT are the optional arguments for the `directory-files'
-function, while IGNORED is the optional 4th argument used with newer versions
+function, while _ is the optional 4th argument used with newer versions
 of `dirctory-files'."
   (apply orig-func dir full match nosort))
 
 (if (< emacs-major-version 28)
     (advice-add 'directory-files :around #'obsidian--directory-files-pre28))
 
-;;;###autoload
 (defun obsidian-specify-path (&optional path)
-  "Specifies obsidian folder PATH to obsidian-folder variable.
+  "Set `obsidian-directory' to PATH or user-selected directory.
 
-When run interactively asks user to specify the path."
-  (interactive)
-  (->> (or path (read-directory-name "Specify path to Obsidian folder"))
+You most likely want to run `obsidian-change-vault'."

```

Comment:
This function changes the vault path but doesn't update the cache, so the function `obsidian-change-vault` will probably be more useful for most applications.